app/models/mixins/supports_feature_mixin.rb

```diff
@@ -0,0 +1,96 @@
+module SupportsFeatureMixin
+  #
+  # Including this in a model gives you a DSL to make features supported or not
+  #
+  #   class Post
+  #     include SupportsFeatureMixin
+  #     supports :publish
+  #     supports_not :fake, "We keep it real"
+  #     supports :archive do |post|
+  #       post.unsupported[:archive] = "Its too good" if post.featured?
+  #     end
+  #   end
+  #
+  # To make a feature conditionally supported, pass a block to the +supports+ method.
+  # The block receives the instance as an argument.
+  # If you set a key with the name of the feature to +instance.unsupported+ with a reason
+  # then the feature will be unsupported and the reason will be accessible through
+  #
+  #   instance.unsupported[:feature]
+  #
+  # The above allows you to call +support_feature?+ methods on the Class and Instance:
+  #
+  #   Post.supports_publish?                       # => true
+  #   Post.new.supports_publish?                   # => true
+  #   Post.supports_fake?                          # => false
+  #   Post.supports_archive?                       # => true
+  #   Post.new(featured: true).supports_archive?   # => false
+  #
+  # To get a reason why a feature is unsupported use the +unsupported+ method
+  #
+  #   Post.unsupported[:publish]                   # => nil
+  #   Post.unsupported[:fake]                      # => "We keep it real"
+  #   Post.new(featured).unsupported[:archive]     # => "Its too good"
```

Comment:
Somehow this Hash interface for `unsupported` feels odd.

Why should the caller know it is a hash?  Caller probably wants to call it as a method: `Post.unsupported(:publish)`.  

Additionally, maybe slightly changing the name of this method to `unsupported_reason` may make it clearer.